shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -434,7 +434,7 @@ private void doInitialFlutterViewRun() {
   private String maybeGetInitialRouteFromIntent(Intent intent) {
     if (host.shouldHandleDeeplinking()) {
       Uri data = intent.getData();
-      if (data != null && !data.getPath().isEmpty()) {
+      if (data != null && data.getPath() != null && !data.getPath().isEmpty()) {
         String fullRoute = data.getPath();
```

Comment:
Why not move this outside the if and use it in the if statement? Instead of calling getPath two times.